tests/model_fields/test_filefield.py

```diff
@@ -54,3 +55,26 @@ def test_refresh_from_db(self):
     def test_defer(self):
         Document.objects.create(myfile='something.txt')
         self.assertEqual(Document.objects.defer('myfile')[0].myfile, 'something.txt')
+
+    def test_unique_when_same_filename(self):
+        """
+        A FileField with the unique attribute enabled should not allow two instances with the same filename to be
+        saved.
+        """
+        d = UniqueDocument(myfile='something.txt')
```

Comment:
use `.objects.create` rather than .save()` (works the same, it's just a bit shorter)